partial-semigroup/src/Data/PartialSemigroup.hs

```diff
@@ -366,7 +366,7 @@ instance PartialSemigroup a => Semigroup (Partial a)
     Partial (Just x) <> Partial (Just y) = Partial (x <>? y)
     _ <> _ = Partial Nothing
 
-instance Monoid a => Monoid (Partial a)
+instance (PartialSemigroup a, Monoid a) => Monoid (Partial a)
```

Comment:
What is this constraint for?

Reply:
In GHC 8.4 and above, `Semigroup` is a superclass of `Monoid`. That means when we make this `Monoid` instance we need to have a `Semigroup` instance, and the `Semigroup` instance for `Partial a` has a constraint on `PartialSemigroup a`. Adding this constraint to the `Monoid` instance discharges that constraint.

But now that you mention it, I'm pretty suspicious of the `Monoid` instance for `Partial`. Since it uses the underlying `Monoid` to just mappend the two things together rather than using the `PartialSemigroup` like the `Semigroup` instance does. Is there a law stating that `PartialSemigroup` must do the same thing as `Semigroup` (and therefore `Monoid`) if they're both defined? I think this definition takes advantage of such a law without it having been stated.

Reply:
Oh, I'm starting to remember. I think the rationale for not defining `PartialMonoid` went something like this: If `i <>? x = Just x` and `x <>? i = Just x` for all `x`, then I can't see any situation in which you don't just have a full `Monoid`.

I'm not sure I ever explicitly specified that there's a law relating `PartialSemigroup` to `Semigroup`, but that's definitely what I had in mind.

Anyway, I'm leaning toward removing the instance.